### ISSUE TITLE
FSLeyes plugin does not install its own venv correctly

### ISSUE THREAD

**Issue by dpapp86** [shimming-toolbox/shimming-toolbox#349]
I have been trying to install the fsleyes plugin, to run B1+ shimming via the GUI. Sadly, there is some venv clash. I suspect I have some trailing python/conda/etc issue somewhere, but I do not know how to fix it.

**###Procedure**

1, I followed the installation documentation here:

`https://shimming-toolbox.org/en/latest/getting_started/installation_gui.html`

The directory from which the installation itself was ultimately run was: 

```
(base) danielpapp@Daniels-MacBook-Pro fsleyes-plugin-shimming-toolbox % pwd
/Users/danielpapp/fsl_plugin_install/fsleyes-plugin-shimming-toolbox
```

2, @gaspardcereza was helping me debug, so we added a flag to the make install to force a clean install:

`(base) danielpapp@Daniels-MacBook-Pro fsleyes-plugin-shimming-toolbox % make install CLEAN=true`

3, The complete output of the installation is in this [pastebin](https://pastebin.com/3vXYiUE7). 

**###Error**

After opening a new terminal, I receive the following error:

``(base) danielpapp@Daniels-MacBook-Pro fsleyes-plugin-shimming-toolbox % shimming-toolbox
Could not find conda environment: pst_venv
You can list all discoverable environments with `conda info --envs`
``

My conda environment list returns:

```
# conda environments:
#
base                  *  /Users/danielpapp/miniconda
dcm2bids                 /Users/danielpapp/miniconda/envs/dcm2bids
                         /Users/danielpapp/shimming-toolbox/python
st_venv                  /Users/danielpapp/shimming-toolbox/python/envs/st_venv
                         /usr/local/fsl/fslpython
                         /usr/local/fsl/fslpython/envs/fslpython
```

**###Investigations and fix attempts**

1, I was curious is this (base) issue, perhaps related to the previous issue here #342 was related, but as far as I can tell, it is not:

```
(base) danielpapp@Daniels-MacBook-Pro fsleyes-plugin-shimming-toolbox % conda deactivate
danielpapp@Daniels-MacBook-Pro fsleyes-plugin-shimming-toolbox % shimming-toolbox
Could not find conda environment: pst_venv
You can list all discoverable environments with `conda info --envs`.

```

2, Note that this is not my first time running this (I had closed the terminals before and wanted to give a full output pastebin). During previous installations, I remember an issue where, during the virtual environment creation, it wanted to write a file, and returned an error because it could not. I do not know why this error is not present now.

The file in question is located here:
`/Users/danielpapp/.conda/environments.txt`

And has the following content, which does not match my "conda env list" output.
```
/usr/local/fsl/fslpython
/usr/local/fsl/fslpython/envs/fslpython
/Users/danielpapp/shimming-toolbox/python
/Users/danielpapp/shimming-toolbox/python/envs/st_venv
```

3, According to the installation log, the environment should be in 
` environment location: /Users/danielpapp/shimming-toolbox/python/envs/st_venv/pst_venv
`

However, it does not seem to be installed there:
```
(base) danielpapp@Daniels-MacBook-Pro st_venv % ls
bin		include		share
conda-meta	lib		ssl
```

Nor does there seem to be a "pst_venv" directory on my Mac, at all.

I suspect this is another one of those "Daniel installed some funky python stuff two years ago for no reason and now there are downsteam issues" thing, and the solution may be to just cleanse anything python-related from my HDD, but hopefully there is a better option.

**Comment by gaspardcereza** [comment]
>3, According to the installation log, the environment should be in
>environment location: /Users/danielpapp/shimming-toolbox/python/envs/st_venv/pst_venv

I don't think it should... It should be located in `/Users/danielpapp/shimming-toolbox/python/envs/pst_venv`, with the exact same structure as `st_venv`. 

I really don't understand why your environments appear to be nested like this...  

I would also precise that you use Catalina, just like me, and that the install works properly on my end, even after deleting all pre-existing conda environments...

**Comment by po09i** [comment]
This is odd.. Can you try:
- Manually deleting `/Users/danielpapp/shimming-toolbox`
- Run `conda env list` and output it here
- Retry the installation normally using `make install`
- Output another pastebin of the installation

**Comment by dpapp86** [comment]
> This is odd.. Can you try:
> 
> * Manually deleting `/Users/danielpapp/shimming-toolbox`

Done

> * Run `conda env list` and output it here

```
(base) danielpapp@Sansfil-Securise-Employes-Poly-230-194 ~ % conda env list
# conda environments:
#
base                  *  /Users/danielpapp/miniconda
dcm2bids                 /Users/danielpapp/miniconda/envs/dcm2bids
                         /usr/local/fsl/fslpython
                         /usr/local/fsl/fslpython/envs/fslpython
```



> * Retry the installation normally using `make install`

Done. I also deleted the original git directory and cloned it again, in case there was something off in that.

> * Output another pastebin of the installation
https://pastebin.com/UnBqw0Eq

The issue is still present:

```
(base) danielpapp@Sansfil-Securise-Employes-Poly-230-194 fsleyes-plugin-shimming-toolbox % shimming-toolbox
Could not find conda environment: pst_venv
You can list all discoverable environments with `conda info --envs`.

(base) danielpapp@Sansfil-Securise-Employes-Poly-230-194 fsleyes-plugin-shimming-toolbox % conda env list  
# conda environments:
#
base                  *  /Users/danielpapp/miniconda
dcm2bids                 /Users/danielpapp/miniconda/envs/dcm2bids
                         /Users/danielpapp/shimming-toolbox/python
st_venv                  /Users/danielpapp/shimming-toolbox/python/envs/st_venv
                         /usr/local/fsl/fslpython
                         /usr/local/fsl/fslpython/envs/fslpython
```

**Comment by po09i** [comment]
The problem is related to this line in the create_venv for the fsleyes plugin:
```
python/bin/conda create -y -n $VENV python=3.7
```

The default environment is not located where we expect it to (inside envs/st_venv/pst_venv instead of envs/pst_venv). Conda has another option to specify the full path of the conda venv.

During a meeting, we used the `-p` option when creating the pst_venv to specify the full path of the location of the venv instead of relying on the default conda location for venv. The above line became:

```
python/bin/conda create -y -p $ST_DIR/$PYTHON_DIR/envs/$VENV python=3.7
```

That solved the problem. I opened an [issue](https://github.com/shimming-toolbox/fsleyes-plugin-shimming-toolbox/issues/28) in the plugin repo to fix it.
